core-bundle/src/EventListener/DbafsMetadataSubscriber.php

```diff
@@ -119,7 +119,7 @@ private function getDefaultLocales(): array
             return $this->defaultLocales;
         }
 
-        $page = $this->requestStack->getMainRequest()?->attributes->get('pageModel');
+        $page = $this->requestStack->getCurrentRequest()?->attributes->get('pageModel');
```

Comment:
This will return an ID in a subrequest, is that intended?

Reply:
Wouldn’t this be converted by the `PageAccessListener` so that it is always a model and not an ID?

Reply:
So only the main request hast the model? Then we should probably use `getMainRequest()` here as well.

Reply:
However, we are using `->attributes->get('pageModel')` on the current request a lot, too:

https://github.com/contao/contao/blob/c10cecec52ed9a1be6c4cf45e8dec6830514f93a/core-bundle/src/Controller/AbstractFragmentController.php#L60-L64

Are you sure it has to be the main request?

Reply:
@ausi is right, we have a listener that converts it (in Contao 5) 😎